src/backend/distributed/executor/multi_utility.c

```diff
@@ -1035,6 +1121,43 @@ PlanAlterTableStmt(AlterTableStmt *alterTableStatement, const char *alterTableCo
 				constraint->skip_validation = true;
 			}
 		}
+#if (PG_VERSION_NUM >= 100000)
+		else if (alterTableType == AT_AttachPartition)
+		{
+			PartitionCmd *partitionCommand = (PartitionCmd *) command->def;
+
+			/*
+			 * We only support ALTER TABLE ATTACH PARTITION, if it is only subcommand of
+			 * ALTER TABLE. It was already checked in ErrorIfUnsupportedAlterTableStmt.
+			 */
+			Assert(list_length(commandList) <= 1);
+
+			rightRelationId = RangeVarGetRelid(partitionCommand->name, NoLock, false);
+
+			/*
```

Comment:
We already return for non-supported cases. See [here](https://github.com/citusdata/citus/pull/1509/files#diff-a8033caa333591e68f4fb7a599757296R1111) and [here](https://github.com/citusdata/citus/pull/1509/files#diff-a8033caa333591e68f4fb7a599757296R1172)